### ISSUE TITLE
How is the abundance calculated for each functional profile?

### ISSUE THREAD

**Issue by aimirza** [xiaoli-dong/metaerg#23]
May you explain how the count and abundance are calculated for each protein-coding gene function profile? For example, how are counts and abundance calculated for MetaCyc pathways? Especially considering that usually not all of the genes are present in the pathways. @Finesim97 said in issue #15:

> if you look for usages of the sample2depth and mdepth_values values in the search and calculated values the depth values are just added together for each step.

However, the codes are hard to follow and I am having a hard time understanding what you mean exactly. Please elaborate. I cant perform statistical tests if I don't know how the abundance was generated. Many thanks.

Ali

**Comment by Finesim97** [comment]
Hi, 
Hopefully this helps:
The depth file is read here:
https://github.com/xiaoli-dong/metaerg/blob/50708247159ad9ed6c05cfa3f75cbdd1d127b8c4/bin/metaerg.pl#L666-L702
The file format was described already in [this issue](https://github.com/xiaoli-dong/metaerg/issues/16#issuecomment-583819438). For every contig the `totalAvgDepth` and the sample names as the keys and the according depth values are stored. 

https://github.com/xiaoli-dong/metaerg/blob/50708247159ad9ed6c05cfa3f75cbdd1d127b8c4/bin/metaerg.pl#L119-L127
This stores the keys (`totalAvgDepth`, sample names) and the values from the contig for each features (rows in GFF). 

The values are reread and stats["cds"]["SAMPLENAME"] stores the total depth overall (each read feature depth is added) and sample2depth["SAMPLENAME"] and is only used for each feature
https://github.com/xiaoli-dong/metaerg/blob/3adfd4151d5c8af859fa34c2d119b731fa6ea833/bin/output_reports.pl#L1027-L1036

The values in sample2depth are used for the profiles and added to the subtypes (pathway id for example): 
https://github.com/xiaoli-dong/metaerg/blob/3adfd4151d5c8af859fa34c2d119b731fa6ea833/bin/output_reports.pl#L1193-L1203
This results in a summed depth value.
This method call creates the file:
https://github.com/xiaoli-dong/metaerg/blob/3adfd4151d5c8af859fa34c2d119b731fa6ea833/bin/output_reports.pl#L1244

In the method the totals are read back:
https://github.com/xiaoli-dong/metaerg/blob/3adfd4151d5c8af859fa34c2d119b731fa6ea833/bin/output_reports.pl#L1497-L1502

For each entry describing info (names, id, etc.) is added to `cols` and then the following calculations are performed:

https://github.com/xiaoli-dong/metaerg/blob/3adfd4151d5c8af859fa34c2d119b731fa6ea833/bin/output_reports.pl#L1569-L1573

This was for the totalAverageDepth, the same thing is done for the different samples in the depth file:
https://github.com/xiaoli-dong/metaerg/blob/3adfd4151d5c8af859fa34c2d119b731fa6ea833/bin/output_reports.pl#L1582-L1590

This means:
1. Each feature gets the depth of the contig (average and samples) assigned
2. For each entry in a profile, these depth values are summed and the hits counted
3. The count is divided by the total count of hits to any type (pathways) in the genome (`Count`, `Count_pct`)
4. The total average and sample depths sums for each pathway is divided by the sum of these values for all pathways (`Abund`, `Abund_pct`, `Abund_SAMPLE`, `Abund_pct_SAMPLE`) 

Therefore the values are just the average of the values for the CDS (contig values), which were annotated with this pathway by MinPath.

**Comment by aimirza** [comment]
Great, thank you for the detailed response!

**Comment by mbcarbonetto** [comment]
Hi! First of all I would like to thank you for the developement of  metaerg. You are making our lives easier.
Following aimirza 's questions, it is still not clear for me what information we get from the Abund column. Could you explain further?
thanks!

**Comment by Ankita-1211** [comment]
@aimirza I just need your help in this. I was using metaerg for one of my project , but never got the end results, i have to manually run each step . I would like to know what is the best practice while installing it through docker. Because docker never worked for me.

**Comment by aimirza** [comment]
> @aimirza I just need your help in this.

Hi @Ankita-1211. I am a user of metaerg, not a developer. Also, I haven't installed it using Docker, I used the instructions in the following link to install: [https://github.com/Finesim97/MetaErgInstallHelper](https://github.com/Finesim97/MetaErgInstallHelper)
